packages/sdk/src/Portfolio/Integrations/Alice.ts

```diff
@@ -0,0 +1,202 @@
+// REFUND ORDER
+//--------------------------------------------------------------------------------------------
+
+export const refundOrder = ExternalPositionManager.makeUse(Action.RefundOrder, refundOrderEncode);
+
+const refundOrderEncoding = [
+  {
+    type: "uint256",
+    name: "orderId",
+  },
+  {
+    type: "uint16",
+    name: "instrumentId",
+  },
+  {
+    type: "bool",
+    name: "isBuyOrder",
+  },
+  {
+    type: "uint256",
+    name: "quantityToSell",
+  },
+  {
+    type: "uint256",
+    name: "limitAmountToGet",
+  },
+  {
+    type: "uint256",
+    name: "timestamp",
+  },
+] as const;
+
+export type RefundOrderArgs = {
+  orderId: bigint;
+  instrumentId: number;
+  isBuyOrder: boolean;
+  quantityToSell: bigint;
+  limitAmountToGet: bigint;
+  timestamp: bigint;
+};
+
+export function refundOrderEncode(args: RefundOrderArgs): Hex {
+  return encodeAbiParameters(refundOrderEncoding, [
+    args.orderId,
+    args.instrumentId,
+    args.isBuyOrder,
+    args.quantityToSell,
+    args.limitAmountToGet,
+    args.timestamp,
+  ]);
+}
+
+export function refundOrderDecode(encoded: Hex): RefundOrderArgs {
+  const [orderId, instrumentId, isBuyOrder, quantityToSell, limitAmountToGet, timestamp] = decodeAbiParameters(
+    refundOrderEncoding,
+    encoded,
+  );
+
+  return {
+    orderId,
+    instrumentId,
+    isBuyOrder,
+    quantityToSell,
+    limitAmountToGet,
+    timestamp,
+  };
+}
+
+//--------------------------------------------------------------------------------------------
+// SWEEP
+//--------------------------------------------------------------------------------------------
+
+export const sweep = ExternalPositionManager.makeUse(Action.Sweep, sweepEncode);
+
+const sweepEncoding = [
+  {
+    type: "uint256[]",
+    name: "orderIds",
+  },
+] as const;
+
+export type SweepArgs = {
+  orderIds: Array<bigint>;
+};
+
+export function sweepEncode(args: SweepArgs): Hex {
+  return encodeAbiParameters(sweepEncoding, [args.orderIds]);
+}
+
+export function sweepDecode(encoded: Hex): SweepArgs {
+  const [orderIds] = decodeAbiParameters(sweepEncoding, encoded);
+
+  return {
+    orderIds: orderIds as bigint[],
+  };
+}
+
+//--------------------------------------------------------------------------------------------
+// EXTERNAL READ FUNCTIONS
+//--------------------------------------------------------------------------------------------
+
+export function getInstrument(
+  client: PublicClient,
+  args: Viem.ContractCallParameters<{
+    aliceOrderManagerAddress: Address;
+    instrumentId: number;
+  }>,
+) {
+  return readContract(client, {
+    ...Viem.extractBlockParameters(args),
+    abi: parseAbi([
+      "function getInstrument(uint16 _instrumentId, bool _mustBeActive) view returns (uint16 id_, bool enabled_, address base_, address quote_)",
+    ]),
+    functionName: "getInstrument",
+    address: args.aliceOrderManagerAddress,
+    args: [args.instrumentId, false],
```

Comment:
I think it would be best to to pass `mustBeActive` flag as param so it can be decided in a place of use, so it will be a little bit more flexible. 

We usually destructure an response array an return values as an object 
```
const [id, enabled ...] = await readContract(...
return {
id,
enabled
...
}

Reply:
Good point, done